src/manager/subnet.rs

```diff
@@ -0,0 +1,57 @@
+use std::collections::HashMap;
+
+///! IPC node-specific traits.
+use anyhow::Result;
+use async_trait::async_trait;
+use fvm_shared::{address::Address, econ::TokenAmount};
+use ipc_gateway::Checkpoint;
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_subnet_actor::{ConstructParams, JoinParams, Status};
+use serde::{Deserialize, Serialize};
+
+/// Trait to interact with a subnet and handle its lifecycle.
+#[async_trait]
+pub trait SubnetManager {
+    /// Deploys a new subnet actor on the `parent` subnet given as an input and with the
+    /// configuration passed in `ConstructParams`.
+    /// The result of the function is the ID address for the subnet actor from which the final
+    /// subet ID can be inferred.
+    fn create_subnet(parent: SubnetID, from: Address, params: ConstructParams) -> Result<Address>;
```

Comment:
This trait will need `lotus` api to interact with the node, it needs to hold `&self`, or we are initializing the lotus api every time in the call?